### DISCUSSION TITLE
Remove-RegistryKey

### DISCUSSION THREAD

**Discussion by QuiDog64** [PSAppDeployToolkit/PSAppDeployToolkit#672] [Q&A] [answered]
Can this be used with wildcards? I want to delete a HKCU reg key for the Start Menu Layout for all current profiles.

**Comment by codylc** [comment]
Can't do this with a wildcard, but check out `Invoke-HKCURegistrySettingsForAllUsers` in the docs.

You'll create a scriptblock of one or more `Remove-RegistryKey` commands you want to execute against each profile. You'll then use `Invoke-HKCURegistrySettingsForAllUsers` to run that scriptblock. 

Examples are in the docs, but let me know if you need more details.

**Reply by QuiDog64** [reply]
I am actually using Invoke-HKCURegistrySettingsForAllUsers..however, the registry key i need to delete is different for each user as it starts with something like this $de${xxxxxx}$$start.tilegrid$windows.data.curatedtilecollection.tilecollection. So I was hoping Remove-RegistryKey could be used like Remove-Item which does work with wildcard.

[scriptblock]$HKCURegistrySettings = {

Remove-RegistryKey -Key 'HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount\*$start.tilegrid$windows.data.curatedtilecollection.tilecollection' -Recurse -SID $UserProfile.SID
       
        }

        Invoke-HKCURegistrySettingsForAllUsers -RegistrySettings $HKCURegistrySettings

**Comment by SierraEcoBravo** [comment]
Remove-RegistryKey is written using -LiteralPath ( e.g. $null = Remove-Item -LiteralPath $Key -Force -Recurse -ErrorAction 'Stop'). That is why it is not working with wildcards.

**Comment by mjr4077au** [comment, marked as answer]
I've added wildcard support to `Remove-ADTRegistryKey` via https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/commit/b598a33f88e598383411486ab0061d1d6eeaff94. It'll be part of PSAppDeployToolkit 4.1.0.
